scalding-serialization/src/main/scala/com/twitter/scalding/serialization/Serialization.scala

```diff
@@ -53,6 +54,11 @@ trait Serialization[T] extends Equiv[T] with Hashing[T] with Serializable {
    * otherwise the caller should just serialize into an ByteArrayOutputStream
    */
   def dynamicSize(t: T): Option[Int]
+
+  // Override this to provide more efficient
+  def skip(in: InputStream): Try[Unit] = {
```

Comment:
I think we may want `def skip(count: Int, in: InputStream): Try[Unit]`  so in say `List[T]` we can skip the rest of the collection. If `count <= 0` do nothing, and otherwise in the worst case just read and throw them away as you do below.